### PR TITLE
Dev/ssoauth modification

### PR DESCRIPTION
**Pull request by mcjaeger**
modifying support for simple single sign on. fetches the http header fields from the incoming request for users. modification is to have the field key from the sw360.properties file. Small code formatting changes in the files affected.

tested with vagrant

### CODE REVIEW

**Comment by mcjaeger** [comment]
both classes seem work, for example the test class produces the following output:

(there was one mistake with the package designation in the properties file)

 2016-01-26 23:12:29,160 INFO  TestAutoLogin:61 - Received header:
(login) header from request with URL from client: 'http://sw360.siemens.com/web/guest'.
 'host'/'sw360.siemens.com'
 'x-real-ip'/'10.0.2.2'
 'x-forwarded-for'/'10.0.2.2'
 'connection'/'close'
 'accept'/'application/x-ms-application, image/jpeg, application/xaml+xml, image/gif, image/pjpeg, application/x-ms-xbap, application/vnd.ms-excel, application/vnd.ms-powerpoint, application/msword, */*'
 'accept-language'/'de-DE'
 'user-agent'/'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.1; WOW64; Trident/6.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; Media Center PC 6.0; .NET4.0C; .NET4.0E; InfoPath.2; MS-RTC LM 8)'
 'accept-encoding'/'gzip, deflate'
 'dnt'/'1'
 'cookie'/'COOKIE_SUPPORT=true; GUEST_LANGUAGE_ID=en_US; ste_vi=vi%3A1e65d6e4ce9e8d3b478112f68f63163d%7Cvi_fv%3A1439894051454; CFID=0; CFTOKEN=0; ste_survey=survey_all%3A1454839322461%257Cno; AUTH_SESSION_ID_REF=SMS_demchdc3m2a_74ef97::9f7d68eb8262a9f1935037d4489700b8'

 2016-01-26 23:12:29,162 ERROR TestAutoLogin:72 - Portal exception at getUserByEmailAddress(): No User exists with the key {companyId=20155, emailAddress=user@sw360.org}
 com.liferay.portal.NoSuchUserException: No User exists with the key {companyId=20155, emailAddress=user@sw360.org}
        at com.liferay.portal.service.persistence.UserPersistenceImpl.findByC_EA(UserPersistenceImpl.java:4535)
        at com.liferay.portal.service.impl.UserLocalServiceImpl.getUserByEmailAddress(UserLocalServiceImpl.java:2681)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.liferay.portal.spring.aop.ServiceBeanMethodInvocation.proceed(ServiceBeanMethodInvocation.java:115)
        at com.liferay.portal.spring.transaction.DefaultTransactionExecutor.execute(DefaultTransactionExecutor.java:62)
        at com.liferay.portal.spring.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:51)
2016-01-26 23:12:29,163 ERROR TestAutoLogin:84 - Could not fetch user from backend: 'user@sw360.org'.

**Comment by mcjaeger** [comment]
commit shouldbe done with autologin enabled, test class will automatically login with user@sw360.org, which might be annoying.

File: frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/users/SSOAutoLogin.java [hunk 1]
```diff
@@ -60,8 +81,8 @@
             log.info(key + ":" + value);
```
Comment (maximilianhuber) [inline frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/users/SSOAutoLogin.java]: I dont think that this information should be logged (except for debugging purposes, maybe reduce logging level?)

**Comment by maximilianhuber** [comment]
@mcjaeger: i will presumably review this PR later this day (when the box is built).

Btw: didn't you want to set the other hook, since TestAutoLogin will log everyone in as user@sw360.org? maybe comment both lines out?

File: frontend/sw360-portlet/src/main/resources/sw360.properties [hunk 2]
```diff
@@ -7,8 +7,21 @@
 # without any warranty.
 #
 
-backend.url= http://127.0.0.1:8080
+# where the frontend searches the backend
+backend.url= http://127.0.0.1:8085
```
Comment (maximilianhuber) [inline frontend/sw360-portlet/src/main/resources/sw360.properties]: As you already have mentioned: the URL of tomcat on the current single-tomcat-setup is http://127.0.0.1:8080

**Comment by maximilianhuber** [comment]
I can confirm that _"automatic login as user `user@sw360.org` when TestAutoLogin is enabled"_ works

**Pull request merged**